recipes/jbig/all/patches/0001-export.patch

```diff
@@ -0,0 +1,99 @@
+diff --git a/libjbig/jbig.h b/libjbig/jbig.h
+index 214105a..8bceb3b 100644
+--- a/libjbig/jbig.h
++++ b/libjbig/jbig.h
+@@ -254,13 +254,12 @@ struct jbg_dec_state {
+ #   if defined(_VISUALC_)
+ #       pragma warning( disable : 4273 )
+ #   endif
+-#   if !defined(_JBIGLIB_)
+-#       define JBIGEXPORT __declspec(dllimport)
+-#   else
+-#       define JBIGEXPORT __declspec(dllexport)
+-#   endif
++#endif
++
++#ifdef WIN32
++#   define JBIGEXPORT __declspec(dllexport)
```

Comment:
Won't this also export the jbig symbols for libraries using jbig?

Reply:
only if they have JBIGEXPORT defined too.

Reply:
The result of your patch is this:
```
ifdef WIN32
#   define JBIGEXPORT __declspec(dllexport)
#else
#   define JBIGEXPORT 
#endif
```
This will unconditionally set `JBIGEXPORT` to `__declspec(dllexport)`on Windows

Reply:
I think it is easier if you replace:

```
#if defined(_MT) && defined(_DLL) && !defined(_JBIGDLL_) && !defined(_LIB)
#   define _JBIGDLL_
#endif
#if defined(_JBIGDLL_)
#   if defined(_VISUALC_)
#       pragma warning( disable : 4273 )
#   endif
#   if !defined(_JBIGLIB_)
#       define JBIGEXPORT __declspec(dllimport)
#   else
#       define JBIGEXPORT __declspec(dllexport)
#   endif
#else
#   define JBIGEXPORT
#endif
```
with
```
#if defined(_VISUALC_)
#   if defined(jbig_EXPORTS)
#       define SHARED_EXPORT_PREFIX __declspec(dllexport)
#   else
#       define SHARED_EXPORT_PREFIX __declspec(dllimport)
#   endif
#else
#   define SHARED_EXPORT_PREFIX
#endif
#if defined(_JBIGDLL_)
#   if defined(_VISUALC_)
#       pragma warning( disable : 4273 )
#   endif
#   define JBIGEXPORT SHARED_EXPORT_PREFIX
#else
#   define JBIGEXPORT
#endif
```
You will also have to define `_JBIGDLL_` in your custom `CMakeLists.txt` and in `self.cpp_info.defines`.

Reply:
(Replace the first `_VISUALC_` with `WIN32`

Reply:
@madebr I did what you said, but I think for some reason some symbol is not been exported. Do you see the reason?

Reply:
@uilianries see bincrafters/conan-center-index#4

See https://cmake.org/cmake/help/latest/prop_tgt/DEFINE_SYMBOL.html for `jbig_EXPORTS`.
cmake will automatically add a define `name_EXPORTS` for a shared library named `name`.